src/corner_kick/src/containers/Canvas/utils/parseLayer.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * This file contains the logic to parse layer data
+ */
+
+import { BYTES_PER_SHAPE } from 'SRC/constants/canvas';
+import { IShape } from 'SRC/types';
+import { ILayerMessage } from 'SRC/types/canvas';
+import { MalformedShapeException } from 'SRC/utils/exceptions/malformedShapes';
+
+/**
+ * Parses an ArrayBuffer into a LayerMessage.
+ * @throws MalformedShapeException if the ArrayBuffer contains an illegal byte count
+ */
+export const parseLayer = (data: ArrayBuffer): ILayerMessage => {
+    const incomingDataView = new DataView(data);
+
+    // Parse the layer number
+    const layer = incomingDataView.getUint8(0);
+
+    // Parse the flags for the layer
+    const flags = incomingDataView.getUint8(1);
+    const flagArray: boolean[] = new Array(8);
+    for (let i = 0; i < 8; i++) {
+        flagArray[i] = ((flags >> i) & 1) === 1;
+    }
+
+    const incomingSpriteCount = (data.byteLength - 2) / BYTES_PER_SHAPE;
+    if (incomingSpriteCount !== Math.round(incomingSpriteCount)) {
```

Comment:
Oops, so you're expecting 2 bytes of layer number and status, and each of the shapes don't have information about status or layers eh?

Reply:
Nope... Hmm, if I understand correctly, you want to send the entire frame all at once, with every shape describing the layer it belongs to. Honestly, I'm okay with doing that, by changing the flag field in the shape to a layer number field. This approach might simplify the logic.

Reply:
It's all good. I have already made the changes on my side to send an payload of data per layer. It follows what you already have here which is:
- 1 byte of layer number
- 1 byte of flag
followed by shape data of repeating structure:
- texture
- flag (I don't know if this is the same or different from the flag byte from above?)
- x
- y
...